packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.ts

```diff
@@ -174,22 +176,77 @@ export class MultichainBalancesController extends BaseController<
       (balanceUpdate: AccountBalancesUpdatedEventPayload) =>
         this.#handleOnAccountBalancesUpdated(balanceUpdate),
     );
-    // TODO: Maybe add a MultichainAssetsController:accountAssetListUpdated event instead of using the entire state.
-    // Since MultichainAssetsController already listens for the AccountsController:accountAdded, we can rely in it for that event
-    // and not listen for it also here, in this controller, since it would be redundant
+
     this.messagingSystem.subscribe(
-      'MultichainAssetsController:stateChange',
-      async (assetsState: MultichainAssetsControllerState) => {
-        for (const accountId of Object.keys(assetsState.accountsAssets)) {
-          await this.#updateBalance(
+      'MultichainAssetsController:accountAssetListUpdated',
+      async ({ assets }) => {
+        const newAccountAssets = Object.entries(assets).map(
+          ([accountId, { added }]) => ({
             accountId,
-            assetsState.accountsAssets[accountId],
-          );
-        }
+            assets: [...added],
+          }),
+        );
+        await this.#handleOnAccountAssetListUpdated(newAccountAssets);
```

Comment:
Sorry, my suggestion was not clear enough last time, but I think we should just a call to `this.#handleOnAccountAssetListUpdated(assets)` on this callback, and then have all the logic handler on on that function (like for the other subscribers)

But that's ok, it's a nitpick :)

Reply:
Oh i see now! alright, will resolve for now this nit if that's okay and refactor later 🙏